### ISSUE TITLE
return from nested loop doesn't return until loop finishes

### ISSUE THREAD

**Issue by craigbarratt** [newville/asteval#72]
I'm using 0.9.18 on python 3.8.3.  Unless I'm doing something wrong, this code
```
def func():
    loop_cnt = 0
    for i in range(5):
        for k in range(5):
            loop_cnt += 1
        return [i, k, loop_cnt]

func()
```
returns `[4, 4, 25]`, versus `[0, 4, 5]` in native python.

**Comment by newville** [comment]
@craigbarratt Oh, thanks and good catch.  I guess we don't have a test for that (or #73).  I think the fix should not be hard (famous last words?) but I'll try to fix this (and #73) and add tests.

**Comment by craigbarratt** [comment]
Yes - all block execution loops need to return once they see a flow-interrupting statement like break, continue or return.

I recently independently developed an [ast interpreter](https://github.com/custom-components/pyscript) for a plugin for home automation system called HASS, and just discovered yours. Yours is elegantly written. I only started on python a few months ago and it's a good to see another implementation for something I've been working on. In reading your code I can definitely see my code has some missing features, or things I've done wrong or not cleanly.

**Comment by newville** [comment]
@craigbarratt   fixed in 0.9.19.   I can believe that async would be useful for a home automation system, but I think that would be a very complex change for something like `asteval`.  It might be worth looking into, but I have not given it much thought.
